### PR TITLE
add multipart/form-data parsing

### PR DESCRIPTION
**Pull request by lukewilliamboswell**
Adds support for multipart/form-data parsing as described in [IETF RFC 7578 Returning Values from Forms: multipart/form-data](https://datatracker.ietf.org/doc/html/rfc7578)
- Tested on macOS using Chrome and Safari browsers

## demo

This demo shows uploading a `.csv`, parse the `multipart/form-data` request body, saves the file to a temporary location and opens it.

demo-3

### CODE REVIEW

**Comment by Anton-4** [comment]
Thanks @lukewilliamboswell :)
Can you add a minimal example and expect script to test it?

**Comment by lukewilliamboswell** [comment]
> Thanks @lukewilliamboswell :) Can you add a minimal example and expect script to test it?

No problem, can do.

**Comment by lukewilliamboswell** [comment]
@Anton-4 I've added an example that uses this. 

roc-logo-small
```
curl 'http://localhost:8000/' \
  -H 'Content-Type: multipart/form-data; boundary=----WebKitFormBoundarykIHm2BDPibpfMOPG' \
  --data-raw $'------WebKitFormBoundarykIHm2BDPibpfMOPG\r\nContent-Disposition: form-data; name="fileToUpload"; filename="roc-logo-small.png"\r\nContent-Type: image/png\r\n\r\n\r\n------WebKitFormBoundarykIHm2BDPibpfMOPG\r\nContent-Disposition: form-data; name="submit"\r\n\r\nUpload .png Image\r\n------WebKitFormBoundarykIH
m2BDPibpfMOPG--\r\n'
```

**Comment by Anton-4** [comment]
> Would you be able to help me with this?

Sure :)

File: platform/Base64.roc
Comment (Anton-4) [inline platform/Base64.roc:1]: Can we add some tests here?

File: platform/Bytes/Encode.roc [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+        Signed8 _ -> 1
+        Unsigned8 _ -> 1
+        Signed16 _ _ -> 2
+        Unsigned16 _ _ -> 2
+        # Signed32 _ -> 4
+        # Unsigned32 _ -> 4
+        # Signed64 _ -> 8
+        # Unsigned64 _ -> 8
+        # Signed128 _ -> 16
+        # Unsigned128 _ -> 16
```
Comment (Anton-4) [inline platform/Bytes/Encode.roc:39]: Why are these commented out?

**Comment by lukewilliamboswell** [comment]
@Anton-4 I've removed Base64 from the platform. I realised we only need it in the example, so I just import it there.

**Pull request merged**